example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,274 @@
+        Policy<Employee> policy = new Policy<>();
+        for (ExampleUserCacheWarmerFactory user : (List<ExampleUserCacheWarmerFactory>) users) {
+            if (user.getUserId().equals(owner)) {
+                policy.owner(user.userWarm());
+            }
+        }
+        for (String key : resourceRules.keySet()) {
+            policy.resourceLevelRule(key, (Rule<Resource>) createRule(resourceRules.get(key), "resource"));
+        }
+        for (String key : recordRules.keySet()) {
+            policy.recordLevelRule(key, (Rule<Employee>) createRule(recordRules.get(key), "record"));
+        }
+        return new MapEntry<>(createResource(), policy);
+    }
+
+    private Rule<?> createRule(final String rule, final String ruleType) {
+        if ("resource".equalsIgnoreCase(ruleType)) {
+            try {
+                LOGGER.debug("Adding rule {} for rule type {}", rule, ruleType);
+                return (Rule<Resource>) Class.forName(rule).getConstructor().newInstance();
+            } catch (ClassNotFoundException | NoSuchMethodException ex) {
+                LOGGER.error("Error getting class: {}", ex.getMessage());
+            } catch (IllegalAccessException e) {
+                LOGGER.error("Error accessing constructor: {}", e.getMessage());
+            } catch (InstantiationException e) {
+                LOGGER.error("Error instantiating: {}", e.getMessage());
+            } catch (InvocationTargetException e) {
+                LOGGER.error("Invocation Target Exception: {}", e.getMessage());
+            }
+        }
+        if ("record".equalsIgnoreCase(ruleType)) {
+            try {
+                LOGGER.debug("Adding rule {} for rule type {}", rule, ruleType);
+                return (Rule<Employee>) Class.forName(rule).getConstructor().newInstance();
+            } catch (ClassNotFoundException | NoSuchMethodException ex) {
+                LOGGER.error("Error getting class: {}", ex.getMessage());
+            } catch (IllegalAccessException e) {
+                LOGGER.error("Error accessing constructor: {}", e.getMessage());
+            } catch (InstantiationException e) {
+                LOGGER.error("Error instantiating: {}", e.getMessage());
+            } catch (InvocationTargetException e) {
+                LOGGER.error("Invocation Target Exception: {}", e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Resource createResource() {
+        URI normalised = ExampleFileUtil.convertToFileURI(resource);
+        String resource = normalised.toString();
+        if (resource.endsWith(".avro")) {
+            return new FileResource().id(resource).type(Employee.class.getTypeName()).serialisedFormat("avro").parent(getParent(resource));
+        } else {
+            return new DirectoryResource().id(resource).parent(getParent(resource));
+        }
+    }
+
+    private ParentResource getParent(final String fileURL) {
```

Comment:
This really needs to be in common at this point
Resource convertToResource(Path path)
Resource convertToResource(URI uri)